index.js

```diff
@@ -48,7 +48,7 @@ async function fastifyEtag (app, opts) {
       reply.header('etag', etag)
     }
 
-    if (req.headers['if-none-match'] === etag) {
+    if (req.headers['if-none-match'] === etag && replyWith304) {
```

Comment:
```suggestion
    if (replyWith304 && req.headers['if-none-match'] === etag) {
```

a boolean check should be faster than looking up the property.